crates/cxx-qt-gen/src/parser/externcxxqt.rs

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::{
+    parser::signals::ParsedSignal,
+    syntax::{attribute::attribute_find_path, safety::Safety},
+};
+use syn::{Attribute, ForeignItem, ItemForeignMod, Result, Token};
+
+/// Representation of an extern "C++Qt" block
+#[derive(Default)]
+pub struct ParsedExternCxxQt {
+    /// Attributes for the extern "C++Qt" block
+    pub attrs: Vec<Attribute>,
+    /// Whether this block has an unsafe token
+    pub unsafety: Option<Token![unsafe]>,
```

Comment:
Interesting, I didn't know Rust allowed macros within type names.

Reply:
Yeah, the `syn` crate seems to use this a lot internally for building the token type.